src/main/java/org/folio/circulation/domain/validation/ItemStatusValidator.java

```diff
@@ -54,4 +57,22 @@ public ItemStatusValidator(
       .next(p -> refuseWhenItemIs(loanAndRelatedRecords, DECLARED_LOST))
       .next(p -> refuseWhenItemIs(loanAndRelatedRecords, CLAIMED_RETURNED));
   }
+
+  public static Result<LoanAndRelatedRecords> refuseWhenItemIsDeclaredLost(
+    Result<LoanAndRelatedRecords> loanResult) {
+
+    return loanResult.failWhen(
+      l -> succeeded(l.getLoan().getItem().isInStatus(DECLARED_LOST)),
+      l -> singleValidationError(new ValidationError("item is Declared lost",
+         ITEM_ID, l.getLoan().getItem().getItemId())));
+  }
+
+  public static Result<LoanAndRelatedRecords> refuseWhenItemIsClaimedReturned(
+    Result<LoanAndRelatedRecords> loanResult) {
+
+    return loanResult.failWhen(
+      l -> succeeded(l.getLoan().getItem().isInStatus(CLAIMED_RETURNED)),
+      l -> singleValidationError(new ValidationError("item is Claimed returned",
+         ITEM_ID, l.getLoan().getItem().getItemId())));
+  }
```

Comment:
Hmm, it seems this validator is currently closely coupled with the check out process. These changes introduce more general methods where the message is not specific to a process. 

Can these be consolidated in some way, to make the differences in message be controlled by the client? Maybe there could be a constructor parameter for a function that generates the appropriate message?

Reply:
Such a function already exists in the constructor, 

```
  private final Function<String, ValidationErrorFailure> itemStatusErrorFunction;

  public ItemStatusValidator(
    Function<String, ValidationErrorFailure> itemStatusErrorFunction) {
    this.itemStatusErrorFunction = itemStatusErrorFunction;
  }
```

Are you suggesting the addition of another such function the contractor's signature? Or using this one that we have?

Reply:
I'm suggesting that I think all of the functions in this class should be similar, likely with the only variants being the expected status and the error message.

These two new functions appear to be structured differently to the others, and don't seem to take advantage of the existing function (that I hadn't noticed from reviewing the diff).

I suggest that either these two functions be adapted to use the existing function, or if necessary, other changes are made to make the class configurable enough for all of the use cases.

Does that make sense?

Reply:
It does, and good sense